tests/integration/web/webfront_test.py

```diff
@@ -192,3 +195,246 @@ def test_should_render_about_logging_modal(client):
 
     assert response.status_code == 200
     assert 'id="about-audit-logging"' in smart_str(response.content)
+
+
+class TestImportDashboardViews:
+    @staticmethod
+    def _get_file_object(data, name='test.json'):
+        """Helper to create a file-like object from a dictionary"""
+        file_content = json.dumps(data).encode('utf-8')
+        file_obj = BytesIO(file_content)
+        file_obj.name = name
+        return file_obj
+
+    def test_should_render_import_dashboard_modal(self, client):
+        """
+        Tests that calling the import_dashboard_modal view will return a fragment
+        with a form to import a dashboard
+        """
+        url = reverse('import-dashboard-modal')
+        response = client.get(url, follow=True)
+
+        assert 'id="import-dashboard-form"' in smart_str(response.content)
+
+    def test_when_method_is_not_post_then_return_method_not_allowed(self, client):
+        """Tests that import_dashboard only accepts POST requests"""
+        url = reverse('import-dashboard')
+        response = client.get(url)
+        assert response.status_code == 405
+
+    @patch("nav.web.webfront.views.can_modify_navlet", return_value=False)
+    def test_when_user_is_without_permission_then_return_forbidden(
+        self, mock_can_modify, client
+    ):
+        """Tests that users without permission cannot import dashboards"""
+        url = reverse('import-dashboard')
+        response = client.post(url)
+        assert response.status_code == 403
+
+    def test_when_no_file_is_provided_then_return_error(self, client):
+        """Tests that importing without a file returns an error modal"""
+        url = reverse('import-dashboard')
+        response = client.post(url)
+
+        assert 'You need to provide a file' in smart_str(response.content)
+
+    def test_given_a_valid_dashboard_file_then_create_dashboard(
+        self, db, client, valid_dashboard_data
+    ):
+        """Tests that importing a valid dashboard file creates a new dashboard"""
+        file_obj = self._get_file_object(
+            valid_dashboard_data, name='test_dashboard.json'
+        )
+
+        url = reverse('import-dashboard')
+        client.post(url, {'file': file_obj})
+
+        # Verify dashboard was created
+        dashboard = AccountDashboard.objects.get(
+            account_id=valid_dashboard_data['account'],
+            name=valid_dashboard_data['name'],
+        )
+        assert dashboard.num_columns == 3
```

Comment:
This to me does not feel like the test that the dashboard was actually created 

I would suggest doing instead 

```suggestion
        assert AccountDashboard.objects.get(
            account_id=valid_dashboard_data['account'],
            name=valid_dashboard_data['name'],
        ).exists()
```

Reply:
I agree, thanks for pointing this out! Fixed.